src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,48 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> top;
+    private int size;
 
+    public void push(T value) {
+        top = new Node<>(top, value);
+        size++;
     }
```

Comment:
Consider adding documentation for the 'push' method to describe its behavior, for example, that it adds an element at the top of the stack.